### PR TITLE
Simplify `nonzero` function

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -4509,62 +4509,23 @@ def _add_scan_blocked_sum_kernel(dtype):
...
-            }
-        }
-    }
-    """).substitute(name=name, src_dtype=src_dtype, index_dtype=index_dtype)
-    module = compile_with_cache(source)
-    return module.get_function(name)
+cdef _nonzero_kernel_1d = ElementwiseKernel(
+    'T src, S index', 'raw S dst',
+    'if (src != 0) dst[index - 1] = i',
+    '_nonzero_kernel_1d')
```
Comment (toslunar) [inline cupy/core/core.pyx]: Could you be consistent with that the name of `_nonzero_kernel` is `nonzero_kernel` (no leading underscore)?

**Comment by okuta** [comment]
jenkins, test this please.

File: cupy/core/core.pyx (hunk 1 above)
Comment (okuta) [reply cupy/core/core.pyx]: I fixed.

**Comment by toslunar** [comment]
Do you have the comparison of the performances?

**Comment by okuta** [comment]
No, I didn't.

**Comment by toslunar** [comment]
I compared the speed
```
master 2018-07-30 13:24:21 +0900 721486b
22.708032608032227 msec
11.698368072509766 msec
10.406111717224121 msec
10.410943984985352 msec
10.417119979858398 msec
10.344287872314453 msec
10.363167762756348 msec
10.369471549987793 msec
10.360511779785156 msec
10.351776123046875 msec

merge PR #1487
21.07708740234375 msec
10.106880187988281 msec
10.225728034973145 msec
10.153823852539062 msec
10.218400001525879 msec
10.166720390319824 msec
11.520959854125977 msec
10.187775611877441 msec
10.201024055480957 msec
10.367391586303711 msec
```

```
def bench(xp):
    x0 = xp.zeros(4096 * 4096)
    x1 = xp.ones(4096 * 4096)
    x2 = xp.random.randint(2, size=(4096, 4096))
    x3 = xp.arange(4096 * 4096).reshape(4096, 4096) % 3
    time.sleep(1)

    start = cupy.cuda.Event()
    end = cupy.cuda.Event()
    start.record()
    y0 = x0.nonzero()
    y1 = x1.nonzero()
    y2 = x2.nonzero()
    y3 = x3.nonzero()
    end.record()
    end.synchronize()
    t = cupy.cuda.get_elapsed_time(start, end)
    print("{} msec".format(t))
```

**Comment by okuta** [comment]
Thnks!

**Pull request merged**